### ISSUE TITLE
Uncompiled `setSize` with two-dimensional object breaks

### ISSUE THREAD

**Issue by dochvam** [nimble-dev/nimble#1023]
`setSize` errors when resizing a two-dimensional object in newer versions of R. We think this relates to the fact that the matrix class now inherits from array. Where the class of the object used to be "matrix," it's now the two elements "array" "matrix".

The following code reproduces the error I'm seeing:

```
library(nimble)
nf <- nimbleFunction(
        setup = function(){},
        run = function() {
            declare(z, double(2))
            setSize(z, 2, 4)
            return(z)
            returnType(double(2))
        })
nf1 <- nf()
nf1$run()
```

The issue is in setSize() which is in types_numericLists.R. One potential solution is to change the line
`newObj <- array(as(0, oldClass), newDims)` to
`newObj <- array(0, newDims)`

This seems to have the drawback of not preserving the info about the type of each element the way we do for 1d objects. However, since in the 3D+ case oldClass is just "array" with no info about underlying type (& I think this is also true for the R 3.6 "matrix"), I think this solution preserves the exact behavior we had before.

This issue was identified while switching our Travis system to use R 4.0 and is being addressed on the R4.0_test branch.

**Comment by paciorek** [comment]
I think @dochvam fixed this. @paciorek to check on this.

**Comment by paciorek** [comment]
This was fixed in commit 6d88a37a3b.
